api/src/test/java/org/openmrs/EncounterTest.java

```diff
@@ -1397,4 +1397,85 @@ public void hasDiagnosis_shouldReturnFalseIfEncounterDoesNotHaveDiagnosis(){
 		assertFalse(encounter.hasDiagnosis(diagnosis));
 	}
 
+	/**
+	 * @see Encounter#getConditions()
+	 */
+	@Test
+	public void getAllConditions_shouldReturnAllCondition(){
+		Encounter encounter = new Encounter();
+
+		Condition activeCondition = new Condition();
+		activeCondition.setClinicalStatus(ConditionClinicalStatus.ACTIVE);
+		CodedOrFreeText freeText = new CodedOrFreeText();
+		freeText.setNonCoded("Free text");
+		activeCondition.setCondition(freeText);
+		encounter.addCondition(activeCondition);
+
+		Condition voidedCondition = new Condition();
+		voidedCondition.setVoided(true);
+		voidedCondition.setClinicalStatus(ConditionClinicalStatus.HISTORY_OF);
+		CodedOrFreeText freeText01 = new CodedOrFreeText();
+		freeText01.setNonCoded("Free text");
+		voidedCondition.setCondition(freeText01);
+		encounter.addCondition(voidedCondition);
+
+		assertEquals(2,encounter.getAllConditions(true).size());
+	}
+
+	/**
+	 * @see Encounter#getConditions()
+	 */
+	@Test
+	public void getConditions_shouldReturnActiveCondition(){
+		Encounter encounter = new Encounter();
+
+		Condition activeCondition = new Condition();
+		activeCondition.setClinicalStatus(ConditionClinicalStatus.ACTIVE);
+		CodedOrFreeText freeText = new CodedOrFreeText();
+		freeText.setNonCoded("Free text");
+		activeCondition.setCondition(freeText);
+		encounter.addCondition(activeCondition);
+
+		Condition voidedCondition = new Condition();
+		voidedCondition.setVoided(true);
+		voidedCondition.setClinicalStatus(ConditionClinicalStatus.HISTORY_OF);
+		CodedOrFreeText freeText01 = new CodedOrFreeText();
+		freeText01.setNonCoded("Free text");
+		voidedCondition.setCondition(freeText01);
+		encounter.addCondition(voidedCondition);
+		
+		assertEquals(1,encounter.getConditions().size());
+	}
+
+	/**
+	 * @see Encounter#getConditions()
+	 */
+	@Test
+	public void removeConditions_shouldRemoveCondition(){
+		Encounter encounter = new Encounter();
+
+		Condition activeCondition = new Condition();
+		activeCondition.setClinicalStatus(ConditionClinicalStatus.ACTIVE);
+		CodedOrFreeText freeText = new CodedOrFreeText();
+		freeText.setNonCoded("Free text");
+		activeCondition.setCondition(freeText);
+		encounter.addCondition(activeCondition);
+		// The condition should be voided
+		encounter.removeCondition(activeCondition);
+
+
+		Condition voidedCondition = new Condition();
+		voidedCondition.setVoided(true);
+		voidedCondition.setClinicalStatus(ConditionClinicalStatus.HISTORY_OF);
+		CodedOrFreeText freeText01 = new CodedOrFreeText();
+		freeText01.setNonCoded("Free text");
+		voidedCondition.setCondition(freeText01);
+		encounter.addCondition(voidedCondition);
+
+
+		Set<Condition> a = encounter.getConditions();
+		Set<Condition> b = encounter.getAllConditions(true);
+		assertEquals(0,encounter.getConditions().size());
+		assertEquals(2,encounter.getAllConditions(true).size());
+	}
```

Comment:
Let's cleanup this class a little. I believe that 1) it should not extend `BaseContextSensitiveTest` and that 2) the two mocked services are not needed.

Can you confirm those two points?

Then I'd simplify your test so that you setup your encounter with a bunch of conditions in a setup method so that you can work on a reference encounter across all your test cases.